### DISCUSSION TITLE
erro of energy minimization

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#341] [Q&A] [answered]
I perform SSCHA+DFT calculations. After finishing QE calculations, I continue doing energy minimization. But there is an erro as follows:
```
Error, the following force files are missing from the ensemble:
[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19]
Traceback (most recent call last):
  File "/public/home/biglinn/sscha-dft-ktao3/minim.py", line 9, in <module>
    ensemble.load("data_and_job", population = POPULATION, N=20)
  File "/public/home/biglinn/miniconda3/envs/sscha/lib/python3.10/site-packages/sscha/Ensemble.py", line 725, in load
    raise IOError(ERROR_MSG)
OSError:
Error, the following force files are missing from the ensemble:
[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19]
```
It seems that it's the problem of input files. So I want to know what the input files of the code "ensemble.load" are.

**Comment by biglinn** [comment]
In the previous step, namely QE calculations, I collected output of all configurations together. Ther are respectively energies_supercell_population1.dat, forces_population_1.dat and pressures_population_1.dat.
Does energy minimzation read them?

**Reply by biglinn** [reply, marked as answer]
How careless of me! I made mistakes in the file name. 
Names of forces files and stress tensor files should be forces_population1_XX.dat and pressures_population1_XX.dat !

**Comment by biglinn** [comment]
However, I have another question.
In the step of ensemble generation, I impose the acoustic sum rule and the symmetries with "dyn.Symmetrize()". Then there will be some files named dyn_start_population1_XX and dyn_end_population1_XX. 
And dyn_start_population1_XX are the same dynamical matrices as the original ones, just imposing the acoustic sum rule and the symmetries.

So, the point is which set of dynamical matrices the energy minimization is based on.
I respectively try two sets of matrices, dyn_start_population1_XX and original harmonic ones. As follows:

Using dyn_start_population1_XX, the codes are:
```
dyn = CC.Phonons.Phonons("dyn_start_population1_", 4)
ensemble = sscha.Ensemble.Ensemble(dyn, 0)
ensemble.load("data_and_job", population = POPULATION, N=20)
```
Using original harmonic dynamical matrices, the codes are:
```
dyn = CC.Phonons.Phonons("harmonic_dyn/ktao3_harmonic_dyn", 4)
dyn.Symmetrize()
ensemble = sscha.Ensemble.Ensemble(dyn, 0)
ensemble.load("data_and_job", population = POPULATION, N=20)
```

Both of two ways can run successfully. So which way is standard?

**Reply by biglinn** [reply]
And are dyn_end_population1_XX useful?
After energy minimization, I save new dynamical matrices with the code: 
`minim.dyn.save_qe("dyn_mat/sscha_dyn_population{}_".format(POPULATION))`

It seems that dyn_end_population1_XX don't play a role.

**Reply by mesonepigreco** [reply]
Yes they are not used, just saved to store all the information. We will probably clean up the savings in one of the next releases.
